src/core/connection.c

```diff
@@ -503,6 +503,8 @@ QuicConnUninitialize(
     QuicCryptoUninitialize(&Connection->Crypto);
     QuicTimerWheelRemoveConnection(&Connection->Worker->TimerWheel, Connection);
     QuicOperationQueueClear(Connection->Worker, &Connection->OperQ);
+    QuicLossDetectionUninitialize(&Connection->LossDetection);
+    QuicSendUninitialize(&Connection->Send);
```

Comment:
Interesting that the code was working perfectly fine before without un-initializing loss detection and send, and its only when we added a Ref from Stream ---> Connection that deadlocks happen

Reply:
Yeah, was it a bug that got discovered from this PR?

Reply:
Not a bug previously. This already happens on free, but because it holds onto streams, we need to clean up here.